dropshot/tests/test_streaming.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 Oxide Computer Company
+
+//! Test cases for streaming reqeusts.
+
+use dropshot::{endpoint, ApiDescription, HttpError, RequestContext};
+use http::{Method, Response, StatusCode};
+use hyper::{body::HttpBody, Body};
+use hyper_staticfile::FileBytesStream;
+use std::sync::Arc;
+use tokio::io::{AsyncSeekExt, AsyncWriteExt};
+
+#[macro_use]
+extern crate slog;
+
+mod common;
+
+fn api() -> ApiDescription<usize> {
+    let mut api = ApiDescription::new();
+    api.register(api_streaming).unwrap();
+    api
+}
+
+const BUF_SIZE: usize = 8192;
+const BUF_COUNT: usize = 128;
+
+#[endpoint {
+    method = GET,
+    path = "/streaming",
+}]
+async fn api_streaming(
+    _rqctx: Arc<RequestContext<usize>>,
+) -> Result<Response<Body>, HttpError> {
+    let mut file = tempfile::tempfile()
+        .map_err(|_| HttpError::for_bad_request(None, "EBADF".to_string()))
```

Comment:
nit: can we use some other string here?  If I see EBADF anywhere I assume it came from errno, and those are really awful to debug.